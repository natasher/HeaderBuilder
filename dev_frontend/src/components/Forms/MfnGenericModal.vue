<template>
  <modal
    name          = "MfnGenericModal"
    @before-open  = "beforeOpen"
    height        = "auto"
    :classes      = "[ 'info-modal', 'v--modal' ]"
    :clickToClose = "closeOnClick" >

    <div class = "header">

      <div class = "left">
        <div class = "title">{{ title }}</div>
      </div>

    </div>

    <div class = "content"
        v-html = "text" ></div>

    <button
      v-for = "( btn, index ) in buttons"
      @click = "btn.handler"
      class = "mfn-button-generic"
      :key = "index">
        {{ btn.text }}
    </button>

  </modal>
</template>

<script>
export default {

  name: 'MfnGenericModal',

  data () {
    return {
      title        : '',
      text         : '',
      buttons      : [],
      closeOnClick : false
    }
  },

  methods: {
    /**
     * Before open modal hook. If openHook is provided, then set item
     * as currenlty used.
     *
     * @event beforeOpen
     * @type {vue-js-modal}
     * @param {event} passes grid item related data
     */
    beforeOpen ( event ) {
      this.title        = event.params.title
      this.text         = event.params.text
      this.buttons      = event.params.buttons
      this.closeOnClick = event.params.closeOnClick
    },
  }
}
</script>
