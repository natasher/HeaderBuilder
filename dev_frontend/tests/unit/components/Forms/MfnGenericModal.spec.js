import { shallowMount, createLocalVue } from '@vue/test-utils'
import flushPromises   from 'flush-promises'
import merge           from 'lodash.merge'
import VModal          from 'vue-js-modal'
import MfnGenericModal from '../../../../src/components/Forms/MfnGenericModal'

const localVue = createLocalVue()
localVue.use( VModal )

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue,
    setData: {
      title  : 'Test Title',
      text   : 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Impedit debitis eveniet eligendi pariatur?',
      buttons: [
        {
          text: 'hide modal',
          handler: jest.fn()
        },
        {
          text: 'some action',
          handler: jest.fn()
        },
      ]
    }
  }

  return shallowMount( MfnGenericModal, merge( defaultMountingOptions, overrides ))
}

describe('MfnGenericModal.vue', () => {

  // INFO: do NOT test external lib ;)
  test('should match snapshot', async () => {
    const wrapper = createWrapper()
    await flushPromises()

    expect( wrapper.element ).toMatchSnapshot()
  })

})