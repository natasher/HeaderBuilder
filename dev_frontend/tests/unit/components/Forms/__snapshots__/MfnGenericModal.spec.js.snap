// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MfnGenericModal.vue should match snapshot 1`] = `
<modal-stub
  classes="info-modal,v--modal"
  delay="0"
  height="auto"
  maxheight="Infinity"
  maxwidth="Infinity"
  minheight="0"
  minwidth="0"
  name="MfnGenericModal"
  overlaytransition="overlay-fade"
  pivotx="0.5"
  pivoty="0.5"
  width="600"
>
  <div
    class="header"
  >
    <div
      class="left"
    >
      <div
        class="title"
      >
        
      </div>
    </div>
  </div>
   
  <div
    class="content"
  />
   
</modal-stub>
`;
